src/SFML/Graphics/Text.cpp

```diff
@@ -540,4 +573,83 @@ void Text::ensureGeometryUpdate() const
     m_bounds.height = maxY - minY;
 }
 
+
+////////////////////////////////////////////////////////////
+void Text::updateLineOffsets() const
+{
+    // temporary use m_lineOffsets to store widths of each line
+    m_lineOffsets.clear();
+
+    // Precompute the variables needed by the algorithm
+    const bool isBold                                        = m_style & Bold;
+    const auto [whitespaceWidth, letterSpacing, lineSpacing] = getSpacing();
+
+    Vector2f      position;
+    std::uint32_t prevChar = 0;
+    float         maxWidth = 0.f;
+    for (std::size_t i = 0; i < m_string.getSize(); ++i)
+    {
+        const std::uint32_t curChar = m_string[i];
+
+        // Apply the kerning offset
+        position.x += m_font->getKerning(prevChar, curChar, m_characterSize, isBold);
+        prevChar = curChar;
+
+        // Handle special characters
+        switch (curChar)
+        {
+            case U' ':
+                position.x += whitespaceWidth;
+                continue;
+            case U'\t':
+                position.x += whitespaceWidth * 4;
+                continue;
+            case U'\n':
+                position.y += lineSpacing;
+                if (position.x > maxWidth)
+                    maxWidth = position.x;
+                m_lineOffsets.push_back(position.x);
+                position.x = 0;
+                continue;
+        }
+
+        // For regular characters, add the advance offset of the glyph
+        position.x += m_font->getGlyph(curChar, m_characterSize, isBold).advance + letterSpacing;
+    }
+
+    // add final part of the string since last newline as the final line
+    // (this is the entire string if text is single line without newlines)
+    m_lineOffsets.push_back(position.x);
+
+    // convert widths into offsets depending on line alignment
+    for (std::size_t i = 0; i < m_lineOffsets.size(); ++i)
+    {
+        switch (m_lineAlignment)
+        {
+            case LineAlignment::Right:
+                m_lineOffsets[i] = maxWidth - m_lineOffsets[i];
+                break;
+            case LineAlignment::Center:
+                m_lineOffsets[i] = (maxWidth - m_lineOffsets[i]) / 2.f;
+                break;
+            case LineAlignment::Left:
+                m_lineOffsets[i] = 0.f;
+                break;
+        }
+        m_lineOffsets[i] = std::round(m_lineOffsets[i]);
+    }
```

Comment:
```suggestion
    // convert widths into offsets depending on line alignment
    const auto applyLineOffset = [&](float value)
    {
        if (m_lineAlignment == LineAlignment::Left)
            return 0.f;
        
        if (m_lineAlignment == LineAlignment::Center)
            return (maxWidth - lineOffset) / 2.f;
        
        assert(m_lineAlignment == LineAlignment::Right);
        return maxWidth - lineOffset;
    };
        
    for (float& lineOffset : m_lineOffsets)
        lineOffset = std::round(applyLineOffset(lineOffset));
```

Reply:
This is quite an elegant use of more modern features.
However, I'd question why the multiple ifs are in place of a switch. Also, maybe we could test for center and right and default to left (i.e. 0 offset) if those options are not given. Which brings to the point that even the switch should have a default label.